wallet/src/de/schildbach/wallet/service/platform/PlatformSyncService.kt

```diff
@@ -782,9 +791,105 @@ class PlatformSynchronizationService @Inject constructor(
+                            if (cachedItems.find {
+                                    it.txId == txIdAsHash && it.cacheTimestamp > doc.createdAt!! &&
+                                        it.originalPrice != null && it.originalPrice != price
+                                } == null
+                            ) {
+                                log.info("processing TxMetadata: gift card price change: changing price")
+                                giftCard.price = price
+                            }
+                        }
+                        metadata.barcodeValue?.let { barcodeValue ->
+                            metadataDocumentRecord.barcodeValue = barcodeValue
+                            log.info("processing TxMetadata: barcode value change")
+                            if (cachedItems.find {
+                                    it.txId == txIdAsHash && it.cacheTimestamp > doc.createdAt!! &&
+                                        it.barcodeValue != null && it.barcodeValue != barcodeValue
+                                } == null
+                            ) {
+                                log.info("processing TxMetadata: barcode value change: changing value")
+                                giftCard.barcodeValue = barcodeValue
+                            }
+                        }
+                        metadata.barcodeFormat?.let { barcodeFormat ->
+                            metadataDocumentRecord.barcodeFormat = barcodeFormat
+                            log.info("processing TxMetadata: barcode format change")
+                            if (cachedItems.find {
+                                    it.txId == txIdAsHash && it.cacheTimestamp > doc.createdAt!! &&
+                                        it.barcodeFormat != null && it.barcodeFormat != barcodeFormat
+                                } == null
+                            ) {
+                                log.info("processing TxMetadata: barcode value change: changing value")
+                                giftCard.barcodeFormat = BarcodeFormat.valueOf(barcodeFormat)
+                            }
+                        }
+                        metadata.merchantUrl?.let { url ->
+                            metadataDocumentRecord.merchantUrl = url
+                            log.info("processing TxMetadata: merchant url change")
+                            if (cachedItems.find {
+                                    it.txId == txIdAsHash && it.cacheTimestamp > doc.createdAt!! &&
+                                        it.merchantUrl != null && it.merchantUrl != url
+                                } == null
+                            ) {
+                                log.info("processing TxMetadata: merchant url change: changing url")
+                                giftCard.merchantUrl = url
+                            }
+                        }
```

Comment:
There might be an opportunity to reduce copy-pasta here, but I'm not sure yet what it would look like.